### PR TITLE
examples: Error details example

### PR DESCRIPTION
**Pull request by larry-safran**
Show how to use google.rpc.Status returned as the details of io.grpc.Status.

### CODE REVIEW

File: examples/src/main/java/io/grpc/examples/errordetails/DetailErrorSample.java [hunk 1]
```diff
@@ -0,0 +1,199 @@
+/*
+ * Copyright 2016 The gRPC Authors
```
[the remark this answers is not in the document]
Comment (larry-safran) [reply examples/src/main/java/io/grpc/examples/errordetails/DetailErrorSample.java]: fixed

File: examples/src/main/java/io/grpc/examples/errordetails/DetailErrorSample.java [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+import io.grpc.stub.StreamObserver;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import javax.annotation.Nullable;
+
+/**
+ * Shows how to set and read com.google.rpc.Status objects as google.rpc.Status error details.
+ */
+public class DetailErrorSample {
```
[the remark this answers is not in the document]
Comment (larry-safran) [reply examples/src/main/java/io/grpc/examples/errordetails/DetailErrorSample.java]: It is convenient to have client and server both in one class for defining the test data and it isn't designed to run multiple times with different arguments.

Did rename it and restructure it (was originally minor mods from the file in errorhandling).

File: examples/src/main/java/io/grpc/examples/errordetails/DetailErrorSample.java [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+
+  private static final String DEBUG_DESC = "detailed error description";
+
+  public static void main(String[] args) throws Exception {
+    new DetailErrorSample().run();
+  }
+
+  private ManagedChannel channel;
+
+  void run() throws Exception {
```
[the remark this answers is not in the document]
Comment (larry-safran) [reply examples/src/main/java/io/grpc/examples/errordetails/DetailErrorSample.java]: restructured code

File: examples/src/main/java/io/grpc/examples/errordetails/DetailErrorSample.java [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+        // This is com.google.rpc.Status, not io.grpc.Status
+        Status status = Status.newBuilder()
+            .setCode(Code.INVALID_ARGUMENT.getNumber())
+            .setMessage("Email or password malformed")
+            .addDetails(Any.pack(DEBUG_INFO))
+            .build();
+        responseObserver.onError(StatusProto.toStatusRuntimeException(status));
+      }
+    }).build().start();
+    channel = Grpc.newChannelBuilderForAddress(
```
[the remark this answers is not in the document]
Comment (larry-safran) [reply examples/src/main/java/io/grpc/examples/errordetails/DetailErrorSample.java]: Good point

File: examples/src/main/java/io/grpc/examples/errordetails/ErrorDetailsExample.java [hunk 5]
```diff
@@ -0,0 +1,222 @@
...
+      DebugInfo.newBuilder()
+          .addStackEntries("stack_entry_1")
+          .addStackEntries("stack_entry_2")
+          .addStackEntries("stack_entry_3")
+          .setDetail("detailed error info.").build();
+
+  public static void main(String[] args) throws Exception {
+    Server server = null;
+    ManagedChannel channel = null;
+    ErrorDetailsExample errorDetailsExample = new ErrorDetailsExample();
```
Comment (temawi) [inline examples/src/main/java/io/grpc/examples/errordetails/ErrorDetailsExample.java:64]: Now ErrorDetailsExample has no state so no point in instantiating it. All the methods are passed in everything they need so you can just make them static.

**Review by temawi — APPROVED** [review]
Look at the one comment I left, but besides that LGTM.

File: examples/src/main/java/io/grpc/examples/errordetails/ErrorDetailsExample.java (hunk 5 above)
Comment (larry-safran) [reply examples/src/main/java/io/grpc/examples/errordetails/ErrorDetailsExample.java]: Done

**Pull request merged**